lib/matplotlib/axes/_axes.py

```diff
@@ -5678,7 +5681,7 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
         #hist_kwargs = dict(range=range, normed=bool(normed))
         # We will handle the normed kwarg within mpl until we
         # get to the point of requiring numpy >= 1.5.
-        hist_kwargs = dict(range=bin_range)
+        hist_kwargs = dict(range=bin_range) if not input_empty else dict()
```

Comment:
The bin_range was (np.inf, -np.inf) when binsgiven was false due to the if not binsgiven block above; I have changed that block to not be entered if input is empty and reverted this change vis  22c6c7f.

Thanks!